projects/vdk-control-cli/tests/vdk/internal/control/command_groups/job/test_download_job.py

```diff
@@ -59,3 +60,24 @@ def test_download_source_dir_exists(httpserver: PluginHTTPServer, tmpdir: LocalP
     ), f"result exit code is not 1, result output: {result.output}"
     # assert it failed for the right reason:
     assert "Directory with name test-job already exists" in result.output
+
+
+def test_download_job_does_not_exist(httpserver: PluginHTTPServer, tmpdir: LocalPath):
+    rest_api_url = httpserver.url_for("")
+    temp_dir = tmpdir.mkdir("foo")
+
+    httpserver.expect_request(
+        uri="/data-jobs/for-team/test-team/jobs/test-job/sources", method="GET"
+    ).respond_with_response(Response(status=404))
+
+    runner = CliRunner()
+    result = runner.invoke(
+        download_job,
+        ["-n", "test-job", "-t", "test-team", "-u", rest_api_url, "-p", temp_dir],
+    )
+
+    assert (
+        result.exit_code == 2
+    ), f"result exit code is not 2, result output: {result.output}"
```

Comment:
Use `test_utils.assert_click_status`  It formats the error message to make it easier to debug failing tests.

Reply:
Good point. Refactored to use test_utils. Thank you